arm9/link.ld

```diff
@@ -4,44 +4,54 @@ ENTRY(_start)
 
 MEMORY
 {
-    VECTORS (RX) : ORIGIN = 0x08000000, LENGTH = 64
     AHBWRAM (RWX) : ORIGIN = 0x08000040, LENGTH = 512K - 64
+    AHBWRAM2 (RWX) : ORIGIN = 0x080a0000, LENGTH = 384K
 }
 
 SECTIONS
 {
-    .vectors : ALIGN(4) {
-        __vectors_lma = LOADADDR(.vectors);
-        __vectors_vma = ABSOLUTE(.);
-        KEEP(*(.vectors));
-        . = ALIGN(4);
-        __vectors_len = ABSOLUTE(.) - __vectors_vma;
-    } >VECTORS AT>AHBWRAM
+    /* this must come *first* so it picks up lua */
+    AHBWRAM2 : ALIGN(4) {
+        build/lua*(.text*);
 
-    .text : ALIGN(4) {
-        __text_s = ABSOLUTE(.);
-        *(.text.start);
-        *(.text*);
         . = ALIGN(4);
-        __text_e = ABSOLUTE(.);
-    } >AHBWRAM
-
-    .rodata : ALIGN(4) {
         *(.rodata*);
-        . = ALIGN(4);
-    } >AHBWRAM
 
-    .data : ALIGN(4) {
-        *(.data*);
+        /* .data */
         . = ALIGN(4);
-    } >AHBWRAM
+        *(.data*);
 
-    .bss : ALIGN(4) {
+        . = ALIGN(4);
         __bss_start = .;
         *(.bss*);
-        . = ALIGN(4);
         __bss_end = .;
-    } >AHBWRAM
 
-    __end__ = ABSOLUTE(.);
+        __end__ = .;
+    } >AHBWRAM2
+
+
+
+    __vectors_lma = 0x08000000;
+    AHBWRAM : ALIGN(4) {
+        /* this needs to be absolute otherwise GM9 won't boot */
+        __vectors_vma = ABSOLUTE(.);
+        KEEP(*(.vectors));
+        . = ALIGN(4);
+        __vectors_len = . - __vectors_vma;
+        . = . + 0x10;
```

Comment:
Why is this extra space required? Just curious

Reply:
This space was present at the end of what used to be `.vectors`.